geopandas/io/arrow.py

```diff
@@ -676,5 +688,58 @@ def _read_feather(path, columns=None, **kwargs):
         raise ImportError("pyarrow >= 0.17 required for Feather support")
 
     path = _expand_user(path)
+
     table = feather.read_table(path, columns=columns, **kwargs)
+    _validate_metadata(table.schema.metadata)
     return _arrow_to_geopandas(table)
+
+
+def _read_parquet_metadata(path, filesystem):
+    ds = import_optional_dependency(
+        "pyarrow.dataset", extra="pyarrow is required for Feather support."
+    )
+    parquet = import_optional_dependency(
+        "pyarrow.parquet", extra="pyarrow is required for Parquet support."
+    )
+
+    metadata = ds.dataset(path, filesystem=filesystem).schema.metadata
+
+    # if GDAL-produced malformed file
+    if metadata is None or b"geo" not in metadata:
+        try:
+            # read_metadata does not accept a filesystem keyword, so need to
+            # handle this manually (https://issues.apache.org/jira/browse/ARROW-16719)
+            if filesystem is not None:
+                pa_filesystem = _ensure_arrow_fs(filesystem)
+                with pa_filesystem.open_input_file(path) as source:
+                    metadata = parquet.read_metadata(source).metadata
```

Comment:
Probably fine for this PR as you just moved this around, but, we just noticing that we should be able to simplify this now, because `read_metadata` has a filesystem keyword for several pyarrow releases